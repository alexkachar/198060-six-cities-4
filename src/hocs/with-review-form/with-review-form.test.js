import React from 'react';
import renderer from 'react-test-renderer';
import PropTypes from 'prop-types';
import {Provider} from 'react-redux';
import withReviewForm from './with-review-form.jsx';
import mockStore from '../../mocks/mock-store.js';

const MockComponent = ({formRef, textRef, submitRef}) => (
  <div>
    <form ref={formRef}>
      <textarea ref={textRef} />
      <button ref={submitRef} />
    </form>
  </div>
);

const WrappedComponent = withReviewForm(MockComponent);

it(`withReviewForm renders correctly`, () => {
  const tree = renderer
    .create(
        <Provider store={mockStore}>
          <WrappedComponent offerId={1} />
        </Provider>, {
          createNodeMock: () => document.createElement(`div`)
        })
    .toJSON();
  expect(tree).toMatchSnapshot();
});

MockComponent.propTypes = {
  formRef: PropTypes.object.isRequired,
  textRef: PropTypes.object.isRequired,
  submitRef: PropTypes.object.isRequired
};
