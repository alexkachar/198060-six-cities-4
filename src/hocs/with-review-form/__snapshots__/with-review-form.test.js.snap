// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withReviewForm renders correctly 1`] = `
<div>
  <form>
    <textarea />
    <button />
  </form>
</div>
`;
